operator-pipeline-images/operatorcert/github.py

```diff
@@ -336,3 +336,81 @@ def close_pull_request(
     """
     pull_request.edit(state="closed")
     return pull_request
+
+
+def copy_branch(
+    github_client: Github,
+    src_repo_name: str,
+    src_branch_name: str,
+    dest_repo_name: str,
+    dest_branch_name: str,
+) -> None:
+    """
+    Copy a branch from Source Repository to Destination Repository.
+
+    Args:
+        github_client (Github): A Github API client
+        src_repo_name (str): The source repository name in the format "organization/repository"
+        src_branch_name(str): The name of the branch to copy
+        dest_repo_name(str): The destination repository name in the format "organization/repository"
+        dest_branch_name(str): The name of the destination branch
+    """
+    src_repository = github_client.get_repo(src_repo_name)
+    dest_repository = github_client.get_repo(dest_repo_name)
+
+    src_branch_ref = src_repository.get_branch(src_branch_name)
+    latest_commit_sha = src_branch_ref.commit.sha
+
+    dest_branch_exists = any(
+        branch.name == dest_branch_name for branch in dest_repository.get_branches()
+    )
```

Comment:
Isn't there any option to do this check without iterating all branches? Maybe getting the dest branch the same way as the src and act upon what it returns? Not sure how the library works in case object does not exist - all the previous calls to get_repo/branch might also need some kind of try/except mechanism to fail correctly if objects don't exist I think.

Reply:
I agree with Tomas' comment about error handling but, since the rest of the module does not have any retry mechanism, I think this can be done as a future improvement.

Also, I don't think we need to handle the case where the destination branch already exists: I would be perfectly ok raising an exception instead of updating the existing branch.